File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/AlumniActAdm/HallOfFrame/HallofframeadmAdp.java
package com.example.adminapp.AlumniActAdm.HallOfFrame;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.adminapp.R;

import java.util.ArrayList;

public class HallofframeadmAdp extends BaseAdapter {


    ArrayList<DataModelHallOfFrame> dataHolder;
    ArrayList<DataModelHallOfFrame> dataHolderBackup;


    public HallofframeadmAdp(ArrayList<DataModelHallOfFrame> dataHolder) {
        this.dataHolder = dataHolder;
        dataHolderBackup = new ArrayList<>(dataHolder);
    }


    @Override
    public int getCount() {
        return dataHolder.size();
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View view, ViewGroup parent) {

        view = LayoutInflater.from(parent.getContext()).inflate(R.layout.single_resource_layout_hallofframe, null);

        // Link those objects with their respective id's
        // that we have given in .XML file
        TextView name = view.findViewById(R.id.imgNameTvSingleResDesHallOfFrame);
        ImageView image = view.findViewById(R.id.imgVSingleResHallOfFrame);

        // Set the data in text view
        name.setText(dataHolder.get(position).getImgNameTvSingleResDesHallOfFrame());

        // Set the image in Image view
        image.setImageResource(dataHolder.get(position).getImgVSingleResHallOfFrame());


        return view;
    }



}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/AlumniActAdm/HallOfFrame/HallofframeadmAdp.kt
package com.example.adminapp.AlumniActAdm.HallOfFrame

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.adminapp.R

class HallofframeadmAdp(var dataHolder: ArrayList<DataModelHallOfFrame>) : BaseAdapter() {
    var dataHolderBackup: ArrayList<DataModelHallOfFrame>

    init {
        dataHolderBackup = ArrayList(
            dataHolder
        )
    }

    override fun getCount(): Int {
        return dataHolder.size
    }

    override fun getItem(position: Int): Any {
        return 0
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, view: View, parent: ViewGroup): View {
        var view = LayoutInflater.from(parent.context)
            .inflate(R.layout.single_resource_layout_hallofframe, null)

        // Link those objects with their respective id's
        // that we have given in .XML file
        val name = view.findViewById<TextView>(R.id.imgNameTvSingleResDesHallOfFrame)
        val image = view.findViewById<ImageView>(R.id.imgVSingleResHallOfFrame)

         val item = dataHolder[position]
        name.text = item.imgNameTvSingleResDesHallOfFrame
        image.setImageResource(item.imgVSingleResHallOfFrame)

        return view
    }
}













